### PR TITLE
LOG-8698: Added support for log management

### PR DESCRIPTION
**Pull request by ellynch-r7**

### CODE REVIEW

**Comment by ellynch-r7** [comment]
This PR is basically identical to the previous one for this branch. I've updated the new commands to be the same style as the refactored ones on master -`lecli <action> <resource>`

File: tests/examples/misc_examples.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 TEST_OWNER_APIKEY = 'test_owner_apikey'
 TEST_OWNER_APIKEY_ID = 'test_owner_apikey_id'
 TEST_ACCOUNT_RESOURCE_ID = 'test_account_resource_id'
+TEST_MANAGEMENT_URL = 'https://rest.logentries.com/management/logs'
```
Comment (stopal-r7) [inline tests/examples/misc_examples.py]: this is meant to be https://rest.logentries.com/management i think.

File: tests/examples/misc_examples.py [hunk 2]
```diff
@@ -36,6 +39,8 @@
 
 MOCK_SAVED_QUERY_URL = 'https://mydummyurl.com/saved_queries'
 
+MOCK_LOGAPI_URL = 'https://mydummyurl.com/management'
```
Comment (stopal-r7) [inline tests/examples/misc_examples.py]: and this should end with /logs

I know that its only for unit tests but for consistency better to fix imho.

File: lecli/log/api.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+
+def handle_get_log_response(response):
+    """
+    Handle get log response
+    """
+    if response_utils.response_error(response):
+        sys.exit(1)
+    elif response.status_code == 200:
+        sys.stdout.write(json.dumps(response.json(), indent=4, sort_keys=True))
```
Comment (stopal-r7) [inline lecli/log/api.py]: can we not use pretty_print_string_as_json util function here?

File: tests/examples/misc_examples.py (hunk 2 above)
Comment (ellynch-r7) [reply tests/examples/misc_examples.py]: Nah, you're totally right.

Fixed now!

File: lecli/log/api.py (hunk 3 above)
Comment (ellynch-r7) [reply lecli/log/api.py]: Yep, will update

File: lecli/api_utils.py [hunk 4]
```diff
@@ -285,3 +294,45 @@ def gensignature(api_key, date, content_type, request_method, query_path, reques
...
+            print_config_error_and_exit(URL_SECTION, 'Log Management URL(%s)' % config_key)
+    except ConfigParser.NoOptionError:
+        print_config_error_and_exit(URL_SECTION, 'Log Management URL(%s)' % config_key)
+
+
+def pretty_print_string_as_json(text):
+    """
+    Pretty prints a json string
+    """
+    print json.dumps(json.loads(text), indent=4, sort_keys=True)
```
Comment (stopal-r7) [inline lecli/api_utils.py:318]: I think we can make this prettier. How you checked tabular library? We are already using it in some other APIs. If it does not work, parsing the response fields is another option. What do you think?

**Review by stopal-r7 — CHANGES_REQUESTED** [review]
Looks good with a possible prettifying of log objects in the response.

File: lecli/api_utils.py (hunk 4 above)
Comment (ellynch-r7) [reply lecli/api_utils.py:318]: I totally misunderstood this on my first read. Looking at changing it now

Comment (stopal-r7) [reply lecli/api_utils.py:318]: sorry the library called "tabulate" - not "tabular".

we are using tabulate in other api's for the moment: https://pypi.python.org/pypi/tabulate
but you can consider using this library as well: https://pypi.python.org/pypi/PTable/0.9.0

**Review by stopal-r7 — APPROVED** [review]
Adding a separator character line between log objects when listing logs can be a good one.

**Review by stopal-r7 — APPROVED** [review]

**Review by stopal-r7 — APPROVED** [review]
when Travis is happy.

**Pull request merged**